src/com/sap/piper/Utils.groovy

```diff
@@ -119,6 +119,7 @@ void pushToSWA(Map parameters, Map config) {
         Telemetry.notify(this, config, parameters)
     } catch (ignore) {
         // some error occured in telemetry reporting. This should not break anything though.
+        echo "[${parameters.step}] Telemetry Report failed: ${ignore.getMessage()}"
```

Comment:
not sure why we should make this explicit ...
I would rather opt for not doing this change

Reply:
In general it is not a good approach to swallow exceptions silently.

>not sure why we should make this explicit ...

We had trouble with a NPE triggered inside `generateSha1`. Reason for that was that in that case `JOB_URL` and `BUILD_URL` was not available (was a fileRunner use case). Was a hard time to find out why there was no telemetry reporting ... For the sake of easy trouble shooting we should not swallow the exception, but make that explicit. Now you might argue: "OK, what about fixing the NPE ?".  That is already done in the meantime. But who knows which exception gets triggered the next time ...

IMO we should have the general habit to not swallow exceptions. In most cases it is fine to let exceptions bubble up. In some cases - where the exception should not break (like here) - it is reasonable to at least log that there was an exception.

> I would rather opt for not doing this change

Not ACK.